docs/versioned_docs/version-zenith/quickstart/292472-arguments.mdx

```diff
@@ -0,0 +1,120 @@
+
+```shell
+dagger -m github.com/shykes/daggerverse/hello call test --giant --name daggernaut
+```
+
+### Pass a container as argument
+
+In addition to basic types, Dagger defines powerful core types, which can be used in a function's arguments. For example, with the core `Container` type, a function can literally receive a container as argument. So, a function that builds a container could use a default base image, but allow passing a custom base as an argument; or a custom publishing function could take a container as argument and upload it to a registry.
+
+This feature is most powerful when calling a function from your own code (see [the section on creating a module](./428201-custom-modules.mdx)). But you can also pass a container argument from the command-line. To do so, add the corresponding flag, followed by the address of an OCI image. The CLI will dynamically pull the image, and pass the resulting container object as argument to the function.
+
+To see how this works, call the `hello` function again, but with a custom container in which to execute the `figlet` tool. Make sure that the container has the tool already installed.
+
+```shell
+dagger -m github.com/shykes/daggerverse/hello call hello --giant --figlet-container index.docker.io/hairyhenderson/figlet
+```
+
+You should see the image being pulled, then the following output:
+
+```shell
+✔ dagger call hello [1.67s]
+┃  _   _      _ _                             _     _ _
+┃ | | | | ___| | | ___    __      _____  _ __| | __| | |
+┃ | |_| |/ _ \ | |/ _ \   \ \ /\ / / _ \| '__| |/ _` | |
+┃ |  _  |  __/ | | (_) |   \ V  V / (_) | |  | | (_| |_|
+┃ |_| |_|\___|_|_|\___( )   \_/\_/ \___/|_|  |_|\__,_(_)
+┃                     |/
+```
+
+It worked! What happens if you pass a container that doesn't have the required tool installed?
+
+```shell
+dagger -m github.com/shykes/daggerverse/hello call hello --giant --figlet-container index.docker.io/alpine
+```
+
+You should see an error telling you that `figlet` could not be found:
+
+```shell
+✘ dagger call hello ERROR [0.98s]
+[...]
+✘ stdout ERROR [0.13s]
+✘ exec figlet Hello, world! ERROR [0.11s]
+┃ exec: "figlet": executable file not found in $PATH
+```
+
+### Pass a directory as argument
+
+Another example is the `Directory` type, representing a directory. To see how this works, use the following command:
+
+```shell
+dagger -m github.com/vikram-dagger/daggerverse/fileutils call tree --dir .
+```
+
+This example uses a file utility module and calls its `Tree()` function. The `Tree()` function accepts a directory (here, the current working directory) as argument and returns a tree representation of that directory. An abbreviated example of the output is shown below:
+
+```shell
+✔ dagger call tree [31.01s]
+┃ .
+┃ ├── README.md
+┃ ├── cli
+┃ │   └── git-build
+┃ │       ├── README.md
+┃ │       └── build.sh
+┃ └── sdk
+┃     ├── go
+┃     │   ├── aws-cdk
+┃     │   │   ├── README.md
+...
+┃ 39 directories, 125 files
+```
```

Comment:
```suggestion
```

You can also pass a remote git reference, and the Dagger CLI will convert it to a `Directory` type referencing the contents of that repository. For example, let's try listing the source code for the Dagger CLI, from the main branch of the Dagger repository:

```shell
$ dagger -m github.com/vikram-dagger/daggerverse/fileutils call tree --dir https://github.com/dagger/dagger#main:cmd/dagger
```

You should be the same file listing as [this github page](https://github.com/dagger/dagger/tree/main/cmd/dagger):

```shell
✔ dagger call tree [1.15s]
┃ .                                                                                              
┃ ├── call.go                                                                                    
┃ ├── cloud.go                                                                                   
┃ ├── debug.go                                                                                   
┃ ├── engine.go                                                                                  
┃ ├── exec_nonunix.go                                                                            
┃ ├── exec_unix.go                                                                               
┃ ├── flags.go                                                                                   
┃ ├── functions.go                                                                               
┃ ├── gen.go                                                                                     
┃ ├── licenses.go                                                                                
┃ ├── listen.go                                                                                  
┃ ├── log.go                                                                                     
┃ ├── main.go                                                                                    
┃ ├── module.go                                                                                  
┃ ├── module_test.go                                                                             
┃ ├── query.go                                                                                   
┃ ├── run.go                                                                                     
┃ ├── session.go                                                                                 
┃ ├── shell.go                                                                                   
┃ └── version.go                                                                                 
┃                                                                                                
┃ 0 directories, 20 files                                                                        
⧗ 3.09s ✔ 79 ∅ 7
```